src/index.js

```diff
@@ -327,6 +335,127 @@ class DisciplCore {
     return { [currentDid]: channelData }
   }
 
+  /**
+   * Exports linked claim data as a W3C Verifiable Presentation starting with the claim the given link refers to.
+   * Links contained in the data of the claim are exported also in a value alongside of the link and links in data of those claims are processed in a same way too etc.
+   * By default, expansion like this is done at most three times. You can alter this depth of the export by setting the second argument. If the maximum depth is reached the exported data
+   * will contain the value MAX_DEPTH_REACHED alongside of the link instead of an exported dataset. You can use this method to iteratively expand the dataset using the link that was not followed.
+   * A claim is never exported twice; circulair references are not followed.
+   */
+  async exportVP (didOrLink, exporterSsid = { did: null, privkey: null }, maxdepth = 3, ssid = null, visitedStack = [], withPrevious = false) {
+    const isDidBool = BaseConnector.isDid(didOrLink)
```

Comment:
This method seems to duplicate a lot of the exportLD logic. Is calling exportLD and transforming the result possible, or is there not enough information in the LD export? In case of the latter we might consider extending the result of the LD export

Reply:
Is possible but restructuring the nested vc's from exportLD into a list of vc's in a vp to be returned from exportVP would not be that efficient when a a lot of vc's are exported...

Reply:
Sonar also signals a lot of duplicated code, so I would like to either extract the common parts to a function and call that, or call the export function and do the double pass over the data. I highly doubt it will add that much load, especially compared to the crypto/network traffic that will be done by the connectors to obtain them in the first place.